authentication/views_test.py

```diff
@@ -68,364 +95,343 @@ def mock_email_send(mocker):
-        return result
-
-    yield run_step
-
-
-@pytest.fixture()
-def register_email_not_exists_with_recaptcha_invalid(
-    settings, client, mock_email_send, mock_recaptcha_failure
-):
-    """Yield a function for this step"""
-
-    def run_step(last_result):  # pylint: disable=unused-argument
-        """Run the step"""
-        settings.RECAPTCHA_SITE_KEY = "fake"
-        result = assert_api_call(
-            client,
-            "psa-register-email",
             {
-                "flow": SocialAuthState.FLOW_REGISTER,
-                "email": NEW_EMAIL,
-                "recaptcha": "fake",
+                "flow": SocialAuthState.FLOW_LOGIN,
+                "state": SocialAuthState.STATE_LOGIN_PASSWORD,
+                "extra_data": {"name": self.user.name},
             },
-            {"error-codes": ["bad-request"], "success": False},
-            expect_status=status.HTTP_400_BAD_REQUEST,
-            use_defaults=False,
         )
-        mock_recaptcha_failure.assert_called_once()
-        mock_email_send.assert_not_called()
-        return result
-
-    yield run_step
-
 
-@pytest.fixture()
-def login_password_valid(client, user):
-    """Yield a function for this step"""
-    password = "password1"
-
-    def run_step(last_result):
-        """Run the step"""
-        user.set_password(password)
-        user.save()
-        return assert_api_call(
-            client,
+    @rule(auth_state=consumes(LoginPasswordAuthStates))
+    def login_password_valid(self, auth_state):
+        """Login with an invalid password"""
+        password = "password123"
+        self.user.set_password(password)
+        self.user.save()
+        assert_api_call(
+            self.client,
             "psa-login-password",
             {
-                "flow": SocialAuthState.FLOW_LOGIN,
-                "partial_token": last_result["partial_token"],
+                "flow": auth_state["flow"],
+                "partial_token": auth_state["partial_token"],
                 "password": password,
             },
             {
-                "flow": SocialAuthState.FLOW_LOGIN,
-                "provider": EmailAuth.name,
+                "flow": auth_state["flow"],
                 "redirect_url": NEXT_URL,
                 "partial_token": None,
                 "state": SocialAuthState.STATE_SUCCESS,
             },
             expect_authenticated=True,
         )
 
-    yield run_step
-
-
-@pytest.fixture()
-def login_password_user_inactive(client, user):
-    """Yield a function for this step"""
-    password = "password1"
-
-    def run_step(last_result):
-        """Run the step"""
-        user.is_active = False
-        user.set_password(password)
-        user.save()
+    @rule(target=LoginPasswordAuthStates, auth_state=consumes(LoginPasswordAuthStates))
+    def login_password_invalid(self, auth_state):
+        """Login with an invalid password"""
+        self.user.set_password("password1")
```

Comment:
If I remove this line the test still passes.

Reply:
So, that's expected because the user's password will then be null and that just means that _all_ passwords will be rejected.

Reply:
Ah makes sense